### DISCUSSION TITLE
How to deploy prefect on azure web app?

### DISCUSSION THREAD

**Discussion by rairafaqatali** [PrefectHQ/prefect#4132] [Q&A]
I'm totally new on prefect. I created a workflow and tested it locally. 
Can anyone guide me on how I'll be able to deploy prefect on the azure app service. I searched a lot but couldn't find any help.

**Comment by zanieb** [comment]
Hi! By "deploy prefect" do you mean run your own instance of Prefect Server or just run flows on the app service?

**Reply by rairafaqatali** [reply]
I want achieve similar that I have on local machine by running 
prefect server start

How i can deploy ui part of prefect on app service?

Thanks for your help

**Reply by zanieb** [reply]
This isn't something we directly support. You'll want to take a look at https://docs.microsoft.com/en-us/azure/app-service/quickstart-multi-container and use the docker-compose file generated by `prefect server start`.

**Reply by rairafaqatali** [reply]
@madkinsz I have created the docker-compose file and it's working fine on localhost. here is my docker-compose file:
`version: '3.7'
services:
  apollo:
    command: bash -c "./post-start.sh && npm run serve"
    environment:
      GRAPHQL_SERVICE_HOST: http://graphql
      GRAPHQL_SERVICE_PORT: 4201
      HASURA_API_URL: http://hasura:3000/v1alpha1/graphql
      PREFECT_API_HEALTH_URL: http://graphql:4201/health
      PREFECT_API_URL: http://graphql:4201/graphql/
      PREFECT_SERVER__TELEMETRY__ENABLED: "true"
    depends_on:
    - graphql
    image: prefecthq/apollo:core-0.14.6
    ports:
    - 4200:4200
    restart: always
  graphql:
    command: bash -c "prefect-server database upgrade -y && python src/prefect_server/services/graphql/server.py" 
    environment:
      PREFECT_CORE_VERSION: "UNKNOWN"
      PREFECT_SERVER_DB_CMD: echo 'DATABASE MIGRATIONS SKIPPED'
      PREFECT_SERVER__DATABASE__CONNECTION_URL: [connection string]
      PREFECT_SERVER__HASURA__HOST: hasura
    depends_on:
    - hasura
    image: prefecthq/server:core-0.14.6
    ports:
    - 4201:4201
    restart: always 
  hasura:
    command: "graphql-engine serve"
    environment:
      HASURA_GRAPHQL_DATABASE_URL: [connection string]
      HASURA_GRAPHQL_ENABLE_CONSOLE: 'true'
      HASURA_GRAPHQL_LOG_LEVEL: warn
      HASURA_GRAPHQL_QUERY_PLAN_CACHE_SIZE: 100
      HASURA_GRAPHQL_SERVER_PORT: '3000'
    depends_on:
    - postgres
    image: hasura/graphql-engine:v1.3.0
    ports:
    - 3000:3000
    restart: always
  postgres:
    command:
    - postgres
    - -c
    - max_connections=150
    environment:
      POSTGRES_DB: [db]
      POSTGRES_PASSWORD: [password]
      POSTGRES_USER: [username]
    image: postgres:11
    ports:
    - 5432:5432
    restart: always
  towel:
    command: python src/prefect_server/services/towel/__main__.py
    depends_on:
    - graphql
    environment:
      PREFECT_SERVER__HASURA__ADMIN_SECRET: hasura-secret-admin-secret
      PREFECT_SERVER__HASURA__HOST: hasura
    image: prefecthq/server:core-0.14.6
    restart: always
  ui:
    command: /intercept.sh
    depends_on:
    - apollo
    environment:
      PREFECT_SERVER__APOLLO_URL: http://localhost:4200/graphql/
    image: prefecthq/ui:core-0.14.6
    ports:
    - 80:8080
    restart: always
 
When I deploy this file on the azure web app for containers using docker compose option, it not loaded properly. The issue is with http://localhost:4200/graphql/ url that we are using in ui. 

Can you kindly let me know, how I'll change this url while deploying on azure web app for container ?
